.github/workflows/CI-pr.yml

```diff
@@ -58,6 +58,59 @@ jobs:
             git push
           fi
 
+  check-msrv:
+    name: Check compilation on MSRV toolchain
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: true
+      matrix:
+        os:
+          - ubuntu-latest
+        # - macos-latest
+        # - windows-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: "Fetch base branch"
+        shell: bash
+        run: git fetch origin $GITHUB_BASE_REF --depth 1
+
+      - name: Install mold linker
+        uses: rui314/setup-mold@v1
+        if: ${{ runner.os == 'Linux' }}
+        with:
+          make-default: true
+
+      - name: Get rust version from Cargo.toml
+        shell: bash
+        run: |
+          rust_version=$(yq -oy '.package.rust-version' Cargo.toml)
+          echo "rust_version=${rust_version}" >> $GITHUB_ENV
+
+      - name: Install Rust
+        uses: dtolnay/rust-toolchain@master
+        with:
+          toolchain: "${{ env.rust_version }}"
+
+      - uses: Swatinem/rust-cache@v2.7.0
+        with:
+          save-if: ${{ github.event_name == 'push' }}
+
+      - name: Install cargo-hack
+        uses: baptiste0928/cargo-install@v2.2.0
+        with:
+          crate: cargo-hack
+
+      - name: Deny warnings
+        shell: bash
+        run: sed -i 's/rustflags = \[/rustflags = \[\n    "-Dwarnings",/' .cargo/config.toml
```

Comment:
Since we're supporting Rust 1.75, we can move the lint configuration to the `Cargo.toml` file.

See: https://blog.rust-lang.org/2023/11/16/Rust-1.74.0.html#lint-configuration-through-cargo

As part of this change, we can remove this step (from this and the stable toolchain job), and add this snippet towards the beginning of the workflow file:

```yaml
env:
  RUSTFLAGS: "-D warnings"
```